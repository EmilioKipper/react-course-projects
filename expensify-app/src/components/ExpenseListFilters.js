import React from "react";
import { connect } from "react-redux";
import {
    setTextFilter,
    sortByDate,
    sortByAmount,
    setStartDate,
    setEndDate
} from "../actions/filters";
import { DateRangePicker } from "react-dates";

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };

    onDateChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate);
        this.props.setEndDate(endDate);
    };

    onFocusChange = calendarFocused => {
        this.setState(() => ({
            calendarFocused
        }));
    };

    onTextChange = e => {
        this.props.setTextFilter(e.target.value);
    }

    onSortChange = e => {
        e.target.value === "amount"
            ? this.props.sortByAmount()
            : this.props.sortByDate()
    }
    render() {
        return (
            <div>
                <input
                    type="text"
                    value={this.props.filters.text}
                    onChange={this.onTextChange}
                />
                <select
                    onChange={this.onSortChange}
                >
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDateChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={() => false}
                />
            </div>
        );
    }
}

const mapStateToProps = state => ({
        filters: state.filters
    })


const mapDispatchToProps = dispatch => ({
    setTextFilter: text => dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount: () => dispatch(sortByAmount()),
    setStartDate: startDate => dispatch(setStartDate(startDate)),
    setEndDate: endDate => dispatch(setEndDate(endDate))
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
