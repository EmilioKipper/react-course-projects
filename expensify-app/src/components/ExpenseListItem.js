import React from 'react'
import { Link } from 'react-router-dom'

const ExpenseListItem = ({ id, description, createdAt, amount, dispatch }) => (
	<div>
		<Link to={`/edit/${id}`}>
			<h3>{description}</h3>
		</Link>
		<p>
			{amount} - {createdAt}
		</p>
	</div>
)

export default ExpenseListItem
