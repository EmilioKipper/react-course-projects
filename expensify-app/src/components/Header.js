import React from "react";
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" exact activeClassName="is-active">
      Dashboard
    </NavLink>
    <NavLink to="/create" activeClassName="is-active">
      Create Expense
    </NavLink>
    <NavLink to="/help" activeClassName="is-active">
      Help
    </NavLink>
  </header>
);

export default Header;
