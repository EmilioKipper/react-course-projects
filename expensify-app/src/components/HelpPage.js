import React from "react";

const HelpPage = () => <div>this is help page component</div>;

export default HelpPage;
