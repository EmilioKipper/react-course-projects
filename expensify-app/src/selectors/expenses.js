import moment from "moment";

export default (expenses, { text, sortBy, startDate, endDate }) => {
    return expenses
        .filter(expense => {
            const createdAtMoment = moment(expense.createAt);
            const startDateMatch = startDate
                ? startDate.isSameOrBefore(createdAtMoment, "day")
                : true;

            const endDateMatch = endDate
                ? endDate.isSameOrAfter(createdAtMoment, "day")
                : true;

            const textMatch = expense.description
                .toLowerCase()
                .includes(text.toLowerCase());

            return startDateMatch && endDateMatch && textMatch;
        })
        .sort((a, b) => {
            if (sortBy === "date") {
                // Return newer first -1 = a, 1 = b
                return a.createAt < b.createAt ? 1 : -1;
            } else if (sortBy === "amount") {
                return a.amount > b.amount ? -1 : 1;
            }
        });
};
