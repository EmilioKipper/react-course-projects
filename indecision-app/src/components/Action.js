import React from "react";

const Action = props => (
  <div>
    <button
      onClick={props.handlePick}
      disabled={!props.hasOptions}
      className="big-button"
    >
      What should I do?
    </button>
  </div>
);

export default Action;
