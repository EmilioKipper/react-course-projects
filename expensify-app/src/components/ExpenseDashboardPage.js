import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from './ExpenseListFilters';


const ExpenseDashbordPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpenseDashbordPage;
