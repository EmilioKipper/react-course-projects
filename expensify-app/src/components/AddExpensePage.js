import React from "react";
import ExpenseForm from "./ExpenseForm";
import { connect } from "react-redux";
import { addExpense } from "../actions/expenses";

const AddExpensePage = props => (
    <div>
        <h1>add expense</h1>
        <ExpenseForm
            onSubmit={expense => {
                props.dispatch(addExpense(expense));
                props.history.push('/');
            }}
        />
    </div>
);

export default connect()(AddExpensePage);
