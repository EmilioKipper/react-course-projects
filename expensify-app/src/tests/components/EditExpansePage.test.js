import React from 'react'
import { shallow } from 'enzyme'
import { EditExpensePage } from '../../components/EditExpensePage'
import expenses from '../fixtures/expenses'

let wrapper, removeExpense, editExpense, history

beforeEach(() => {
	removeExpense = jest.fn()
	editExpense = jest.fn()
	history = { push: jest.fn() }
	wrapper = shallow(
		<EditExpensePage
			removeExpense={removeExpense}
			editExpense={editExpense}
            history={history}
            expense={expenses[1]}
		/>
	)
})

test('should render EditExpensePage', () => {
	expect(wrapper).toMatchSnapshot()
})

test('should handle editExpense', () => {
	wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1])
	expect(history.push).toHaveBeenLastCalledWith('/')
	expect(editExpense).toHaveBeenLastCalledWith(expenses[1].id, expenses[1])
})

test('should handle removeExpense', () => {
	wrapper.find('button').simulate('click')
	expect(history.push).toHaveBeenLastCalledWith('/')
	expect(removeExpense).toHaveBeenLastCalledWith({id: expenses[1].id})
})
