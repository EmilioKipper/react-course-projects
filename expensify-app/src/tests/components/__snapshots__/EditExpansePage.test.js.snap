// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render EditExpensePage 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 109500,
        "createdAt": -345600000,
        "description": "Rent",
        "id": "2",
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
